### ISSUE TITLE
Download tarballs are rooted at "." instead of a subdirectory

### ISSUE THREAD

**Issue by BonkaBonka** [jonashaag/klaus#204]
I spent some time collecting results from how various hosted SCMs do archives: https://gist.github.com/BonkaBonka/2d38946f4884ab3409b4c8a63efcac2c

All of them except Klaus place files in a subdirectory.  Ideally, I'd like Klaus to mimic what cgit does with respect to how it names the directory inside the archive.  The heuristic it uses is described in this comment: https://git.zx2c4.com/cgit/tree/ui-snapshot.c?h=v1.1#n132

**Comment by jonashaag** [comment]
Agreed, the cgit heuristic looks solid. We should name the folder exactly like the tarball file name, i.e. `foo-1.0.0.tar.gz` -> `foo-1.0.0`

**Comment by BonkaBonka** [comment]
Can we use a dash instead of an @ to join the repo name to the repo revision?

**Comment by jonashaag** [comment]
Sure; what's the reasoning for you here?

**Comment by BonkaBonka** [comment]
Oh, I was trying to match cgit's behavior.

But it's a bit harder because the fix for #36 ([sanitize_branch_name](https://github.com/jonashaag/klaus/blob/d48655f41d2a5d496d69692d0b95f9310fa5b73f/klaus/utils.py#L235)) also swaps out periods in the tag name with dashes.  I get slashes in tags are bad but am not sure why periods got blacklisted at the same time.

**Comment by jonashaag** [comment]
> I get slashes in tags are bad but am not sure why periods got blacklisted at the same time.

Good question. @lb1a @stpierre @posativ do you recall what the reasoning was here?
